File: SpringBootSIVAP/sivap/src/main/java/br/uel/trabalho/sivap/model/Safra.java
package br.uel.trabalho.sivap.model;

import java.math.BigDecimal;
import java.util.Date;

public class Safra {
    private int id_safra;
    private int id_propriedade;
    private int id_talhao;
    private int id_variedade_cultura;
    private Date dt_plantio;
    private Date dt_colheita;
    private BigDecimal producao;

    public Safra(int id_safra, int id_propriedade, int id_talhao, int id_variedade_cultura, 
                 Date dt_plantio, Date dt_colheita, BigDecimal producao) {
        this.id_safra = id_safra;
        this.id_propriedade = id_propriedade;
        this.id_talhao = id_talhao;
        this.id_variedade_cultura = id_variedade_cultura;
        this.dt_plantio = dt_plantio;
        this.dt_colheita = dt_colheita;
        this.producao = producao;
    }

    public Safra() {
    }

    public int getId_safra() {
        return id_safra;
    }

    public void setId_safra(int id_safra) {
        this.id_safra = id_safra;
    }

    public int getId_propriedade() {
        return id_propriedade;
    }

    public void setId_propriedade(int id_propriedade) {
        this.id_propriedade = id_propriedade;
    }

    public int getId_talhao() {
        return id_talhao;
    }

    public void setId_talhao(int id_talhao) {
        this.id_talhao = id_talhao;
    }

    public int getId_variedade_cultura() {
        return id_variedade_cultura;
    }

    public void setId_variedade_cultura(int id_variedade_cultura) {
        this.id_variedade_cultura = id_variedade_cultura;
    }

    public Date getDt_plantio() {
        return this.dt_plantio;
    }

    public void setDt_plantio(Date dt_plantio) {
        this.dt_plantio = dt_plantio;
    }

    public Date getDt_colheita() {
        return this.dt_colheita;
    }

    public void setDt_colheita(Date dt_colheita) {
        this.dt_colheita = dt_colheita;
    }

    public BigDecimal getProducao() {
        return this.producao;
    }

    public void setProducao(BigDecimal producao) {
        this.producao = producao;
    }
}

File: ReactSIVAP/sivap_ui/src/screens/Safra.js
import { useState } from 'react';
import '../styles/Safra.css';

const Safra = () => {
  const [idPropriedade, setIdPropriedade] = useState('');
  const [idTalhao, setIdTalhao] = useState('');
  const [idVariedadeCultura, setIdVariedadeCultura] = useState('');
  const [dtPlantio, setDtPlantio] = useState('');
  const [dtColheita, setDtColheita] = useState('');
  const [producao, setProducao] = useState('');
  
  const sendForm = async (event) => {
    event.preventDefault();

    if(!idPropriedade || !idTalhao || !idVariedadeCultura || !dtPlantio || !dtColheita || !producao) {
      alert("Preencha todos os campos para enviar os dados");
      return;
    }

    try {
      const response = await fetch('http://localhost:8080/safras', {
        method:'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          id_propriedade: parseInt(idPropriedade),
          id_talhao: parseInt(idTalhao),
          id_variedade_cultura: parseInt(idVariedadeCultura),
          dt_plantio: dtPlantio,
          dt_colheita: dtColheita,
          producao: parseFloat(producao)
        })
      });
      
      if (response.ok) {
        alert("Safra cadastrada com sucesso!");
        setIdPropriedade('');
        setIdTalhao('');
        setIdVariedadeCultura('');
        setDtPlantio('');
        setDtColheita('');
        setProducao('');
      } else {
        alert("Erro ao cadastrar safra");
      }
    }
    catch(error) {
      console.log("erro ao processar request: " + error);
      alert("Erro de conexão com o servidor");
    }
  }

  return (
    <div className="safra-container">
      <div className="safra-left">
        <div className="safra-logo">🌱 AgroSIVAP</div>
        <h1>Cadastrar<br />Safra</h1>
        <p>Registre uma nova safra para controle de produção</p>
        
        <form className="safra-form" onSubmit={sendForm}>
          <div className="form-row">
            <div className="form-group">
              <label>ID da Propriedade:</label>
              <input 
                type="number" 
                value={idPropriedade}
                onChange={(e) => setIdPropriedade(e.target.value)}
                placeholder="Digite o ID da propriedade"
                required
              />
            </div>

            <div className="form-group">
              <label>ID do Talhão:</label>
              <input 
                type="number" 
                value={idTalhao}
                onChange={(e) => setIdTalhao(e.target.value)}
                placeholder="Digite o ID do talhão"
                required
              />
            </div>
          </div>

          <div className="form-row">
            <div className="form-group">
              <label>ID Variedade da Cultura:</label>
              <input 
                type="number" 
                value={idVariedadeCultura}
                onChange={(e) => setIdVariedadeCultura(e.target.value)}
                placeholder="Digite o ID da variedade"
                required
              />
            </div>

            <div className="form-group">
              <label>Produção (kg):</label>
              <input 
                type="number" 
                step="0.01"
                value={producao}
                onChange={(e) => setProducao(e.target.value)}
                placeholder="Digite a produção em kg"
                required
              />
            </div>
          </div>

          <div className="form-row">
            <div className="form-group">
              <label>Data de Plantio:</label>
              <input 
                type="date" 
                value={dtPlantio}
                onChange={(e) => setDtPlantio(e.target.value)}
                required
              />
            </div>

            <div className="form-group">
              <label>Data de Colheita:</label>
              <input 
                type="date" 
                value={dtColheita}
                onChange={(e) => setDtColheita(e.target.value)}
                required
              />
            </div>
          </div>
          
          <button type="submit">Cadastrar Safra</button>
        </form>
      </div>
      
      <div className="safra-right">
        <div className="placeholder-img">
          🌾<br />
          Controle de<br />
          Safras
        </div>
      </div>
    </div>
  );
}

export default Safra;